New sidenav

amcdnl opened the pull request:
This PR implements the new left nav. This is the exact same PR as #300 just merged w/ @mmalerba latest changes.

mmalerba commented:
I see a couple of issues here:
- In mobile mode, the document does not receive scroll events (this is needed for the address bar to scroll up on phones (compare to https://material2-docs-dev.firebaseapp.com/components/autocomplete/overview where the document does receive scroll events)
- In mobile mode, there seems to be a gap between the 2 toolbars
![docs-site1](https://user-images.githubusercontent.com/14793288/32284709-4c7f53d0-bee5-11e7-8cb0-9c4e47b91cba.png)
- In desktop mode, there is a gap at the bottom (below the footer)
![docs-site2](https://user-images.githubusercontent.com/14793288/32284749-67a01a82-bee5-11e7-9718-a07a85e709c7.png)
- In desktop mode, when I expand the sidenav both the content and the sidenav scroll (the content doesn't actually need to scroll, it's only scrolling because of the sidenav) 
![docs-site3](https://user-images.githubusercontent.com/14793288/32284841-bcbe0e52-bee5-11e7-9bd7-8f886b8fe9aa.png)

amcdnl commented:
1) So that is tricky because of the sticky positioned items. I was able to work around this by adding a hack that will automatically hide the browser address bar for you.
2) Fixed
3) Fixed
4) I'm not sure its possible to fix this given the nature of things that need to scroll / push in this scenario. Open to suggestions though.

amcdnl commented:
tag @jelbourn

jelbourn commented:
@mmalerba can you review this?

mmalerba commented:
Sorry, I thought this wasn't ready since it has conflicts. @amcdnl Can you resolve it so I can bring up a demo?

amcdnl commented:
@mmalerba - merged :)

amcdnl commented:
@mmalerba - I believe I had addressed all your feedback and am ready for review! :)

mmalerba commented on src/app/pages/component-sidenav/_component-sidenav-theme.scss:
```diff
@@ -1,36 +1,49 @@
+  $nav-background-opacity: if($is-dark-theme, 0.2, 0.03);
 
+  .docs-component-viewer-nav-content {
+    background: rgba(0, 0, 0, $nav-background-opacity);
```
should this be the foreground color instead of always black?

amcdnl replied:
I wanted to make it a slight lighter accent of the current background. This is mainly done like this for different themes.

mmalerba commented on src/app/pages/component-sidenav/_component-sidenav-theme.scss:
```diff
@@ -1,36 +1,49 @@
 
-  .docs-component-viewer-sidenav {
+    &::-webkit-scrollbar-thumb {
+      background: rgba(0, 0, 0, .26);
```
same comment as above

mmalerba commented on src/app/pages/component-sidenav/_component-sidenav-theme.scss:
```diff
@@ -1,36 +1,49 @@
+
+    .mat-drawer {
+      &::-webkit-scrollbar-thumb {
+        background: rgba(0, 0, 0, .26);
```
same comment as above

mmalerba commented on src/app/pages/component-sidenav/component-nav.html:
```diff
@@ -0,0 +1,23 @@
+      <hr *ngIf="!last" />
+    </nav>
+  </div>
+</div>
```
nit: \n

mmalerba commented on src/app/pages/component-sidenav/component-sidenav.html:
```diff
@@ -1,25 +1,25 @@
+    </div>
   </div>
-</mat-sidenav-container>
+</mat-sidenav-container>
```
nit: \n

mmalerba commented:
Awesome, it looks much better now. It looks like this one is still an issue:

> - In desktop mode, when I expand the sidenav both the content and the sidenav scroll (the content doesn't actually need to scroll, it's only scrolling because of the sidenav) 
![docs-site3](https://user-images.githubusercontent.com/14793288/32284841-bcbe0e52-bee5-11e7-9bd7-8f886b8fe9aa.png)

I don't think it's a huge deal, I would be ok submitting the PR if it's too difficult to fix, but it would be nice to fix.

amcdnl commented:
@mmalerba - The nav is always a fixed 75vh so it should be scrolling in this case. The body scrolling in this case is unexpected and I thought I had fixed this. Let me take another look and let you know.

amcdnl commented:
@mmalerba - Really the only good fix for this would be to remove the scrollbar from the side nav and just let it push down. I implemented the scrollbar like this similar to the google api docs that I used as an example. Let me know how you would like to proceed.

mmalerba commented:
we can't do something like this?

```css
max-height: calc(100vh - <header-height> - <footer-height>)
```

amcdnl commented:
![](https://media2.giphy.com/media/axu6dFuca4HKM/giphy.gif)
